src/MatrixClientPeg.ts

```diff
@@ -273,15 +274,20 @@ class MatrixClientPegClass implements IMatrixClientPeg {
         opts.threadSupport = true;
 
         if (SettingsStore.getValue("feature_sliding_sync")) {
+            const baseUrl = this.matrixClient.baseUrl;
             const proxyUrl = SettingsStore.getValue("feature_sliding_sync_proxy_url");
+            const wellKnown = AutoDiscovery.findClientConfig(baseUrl); // the client isn't init'd yet
+            const wellKnownProxyUrl = (await wellKnown)?.["org.matrix.msc3575.proxy"]?.url;
```

Comment:
Can we have the await on the previous line, neater than brackets and also corrects the value of `wellKnown` as right now its now the wellKnown but instead a promise of one.